src/core/pointcloud/qgspointcloudlayer.cpp

```diff
@@ -959,14 +962,16 @@ void QgsPointCloudLayer::loadIndexesForRenderContext( QgsRenderContext &renderer
     }
 
     const QVector<QgsPointCloudSubIndex> subIndex = mDataProvider->subIndexes();
+    const QgsVirtualPointCloudProvider &vpcProvider = dynamic_cast<QgsVirtualPointCloudProvider &>( *mDataProvider );
```

Comment:
The usage of `averageSubIndexWidth` and `averageSubIndexHeight` has been discussed also with @wonder-sk and we will keep it as it is for now. Just the condition will be changed to your suggestion (eg. on ultra-wide monitors it will be switching to actual point clouds too late because of width). 

There was a notion that in the future we could rework the condition to map scale and users will be able to also set an arbitrary number, when the overview switches.